test/ipcAPI.cpp

```diff
@@ -0,0 +1,336 @@
+    };
+
+    stats *stat = nullptr;
+    umf_test::provider_malloc helper_prov;
+    allocations_mutex_type alloc_mutex;
+    allocations_map_type allocations;
+
+    umf_result_t initialize(stats *s) noexcept {
+        stat = s;
+        return UMF_RESULT_SUCCESS;
+    }
+    enum umf_result_t alloc(size_t size, size_t align, void **ptr) noexcept {
+        auto ret = helper_prov.alloc(size, align, ptr);
+        if (ret == UMF_RESULT_SUCCESS) {
+            allocations_write_lock_type lock(alloc_mutex);
+            auto [it, res] = allocations.emplace(*ptr, size);
+            (void)it;
+            EXPECT_TRUE(res);
+        }
+        return ret;
+    }
+    enum umf_result_t free(void *ptr, size_t size) noexcept {
+        auto ret = helper_prov.free(ptr, size);
+        allocations_write_lock_type lock(alloc_mutex);
+        allocations.erase(ptr);
+        return ret;
+    }
+    const char *get_name() noexcept { return "mock_ipc"; }
+    enum umf_result_t get_ipc_handle_size(size_t *size) noexcept {
+        *size = sizeof(ipc_data);
+        return UMF_RESULT_SUCCESS;
+    }
+    enum umf_result_t get_ipc_handle(const void *ptr, size_t size,
+                                     void *ipcDataOpaque) noexcept {
+        ++stat->getCount;
+        ipc_data *ipcData = static_cast<ipc_data *>(ipcDataOpaque);
+        allocations_read_lock_type lock(alloc_mutex);
+        auto it = allocations.find(ptr);
+        if (it == allocations.end()) {
+            // client tries to get handle for the pointer that does not match
+            // with any of the base addresses allocated by the instance of
+            // the memory provider
+            return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+        }
+        (void)size;
+        ipcData->ptr = ptr;
+        ipcData->size = it->second; // size of the base allocation
+        return UMF_RESULT_SUCCESS;
+    }
+    enum umf_result_t put_ipc_handle(void *ipcData) noexcept {
+        ++stat->putCount;
+        (void)ipcData;
+        return UMF_RESULT_SUCCESS;
+    }
+    enum umf_result_t open_ipc_handle(void *ipcDataOpaque,
+                                      void **ptr) noexcept {
+        ++stat->openCount;
+        ipc_data *ipcData = static_cast<ipc_data *>(ipcDataOpaque);
+        void *mapping = std::malloc(ipcData->size);
+        if (!mapping) {
+            return UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+        }
+
+        memcpy(mapping, ipcData->ptr, ipcData->size);
+
+        *ptr = mapping;
+
+        return UMF_RESULT_SUCCESS;
+    }
+    enum umf_result_t close_ipc_handle(void *ptr) noexcept {
+        ++stat->closeCount;
+        std::free(ptr);
+        return UMF_RESULT_SUCCESS;
+    }
+};
+
+struct umfIpcTest : umf_test::test {
+    umfIpcTest() : pool(nullptr, nullptr) {}
+    void SetUp() override {
+        test::SetUp();
+        this->pool = makePool();
+    }
+
+    void TearDown() override { test::TearDown(); }
+
+    umf::pool_unique_handle_t makePool() {
```

Comment:
You can use this: https://github.com/oneapi-src/unified-memory-framework/blob/main/test/memoryPool.hpp#L23

Perhaps we should move that function to some place where it's more visible.